packages/onboard-helpers/src/hooks/useOnboardHelpers.ts

```diff
@@ -0,0 +1,356 @@
+  const navigate = useNavigate()
+  const location = useLocation()
+
+  const locationPathname = location.pathname
+  const { walletChainId, walletSignerAddress } = getWallet(wallet)
+
+  const isSameChainId = {
+    apiRoute: apiChainId === routeChainId && !isIntegrationPage,
+    apiWallet: !!walletChainId && +walletChainId === apiChainId,
+    routeWallet: !!walletChainId && +walletChainId === routeChainId,
+    apiWalletRoute:
+      !!walletChainId && +walletChainId === apiChainId && +walletChainId === routeChainId && !isIntegrationPage,
+  }
+
+  const isSameAddress = {
+    apiWallet: apiSignerAddress.toLowerCase() === walletSignerAddress,
+  }
+
+  const isSameLocale = routeLocale === document.documentElement.lang
+
+  const updateConnectSwitchNetwork = useCallback(
+    (options: ConnectOptions[CONNECT_STAGE.SWITCH_NETWORK]) => {
+      updateConnectState('loading', CONNECT_STAGE.SWITCH_NETWORK, options)
+    },
+    [updateConnectState]
+  )
+
+  const updateConnectApi = useCallback(
+    (options: ConnectOptions[CONNECT_STAGE.CONNECT_API]) => {
+      updateConnectState('loading', CONNECT_STAGE.CONNECT_API, options)
+    },
+    [updateConnectState]
+  )
+
+  useEffect(() => {}, [connectState.stage, connectState.status])
+
+  const handleReconnectWallet = useCallback(
+    async (walletState: WalletState, chainId?: number) => {
+      const label = walletState.label
+      await disconnect(walletState)
+      const newWalletState = (await connect({ autoSelect: { label, disableModals: true } }))[0]
+      if (chainId) {
+        await setChain({ chainId: ethers.toQuantity(chainId) })
+        await _isNetworkSwitched(chainId)
+      }
+      return newWalletState
+    },
+    [connect, disconnect, setChain]
+  )
+
+  const handleConnectCurveApi = useCallback(
+    async ([chainId, useWallet]: ConnectOptions[CONNECT_STAGE.CONNECT_API]) => {
+      try {
+        let parsedWallet = useWallet ? wallet : null
+        const { walletChainId } = getWallet(parsedWallet)
+
+        // check wallet sync again
+        if (parsedWallet && walletChainId && +walletChainId !== chainId) {
+          parsedWallet = await handleReconnectWallet(parsedWallet, chainId)
+        }
+
+        await updateApi(chainId, useWallet ? parsedWallet : null)
+        updateConnectState('success', '')
+      } catch (error) {
+        console.error(error)
+        updateConnectState('failure', CONNECT_STAGE.CONNECT_API)
+      }
+    },
+    [wallet, updateApi, updateConnectState, handleReconnectWallet]
+  )
+
+  const handleConnectWallet = useCallback(
+    async (walletName: ConnectOptions[CONNECT_STAGE.CONNECT_WALLET]) => {
+      let walletState
+
+      if (!walletName) {
+        walletState = (await connect())?.[0]
+      } else {
+        // If found label in localstorage, after 30s if not connected, reconnect with modal
+        const walletStatesPromise = new Promise<WalletState[] | null>(async (resolve, reject) => {
```

Comment:
Like that it will be more readable
```ts
const walletStatesPromise = Promise.race([
  connect({ autoSelect: { label: walletName, disableModals: true } }),
  new Promise<never>((_, reject) => setTimeout(() => reject(new Error('timeout connect wallet')), 3000)),
])
  .then(walletStates => walletStates)
  .catch(error => Promise.reject(error));
```